lightly/api/download.py

```diff
@@ -222,3 +221,100 @@ def job(*args):
+    
+    Args:
+        url:
+            The url of the video.
+        video_channel:
+            The video stream channel from which to find the number of frames.
+        thread_type:
+            Which multithreading method to use for decoding the video.
+            See https://pyav.org/docs/stable/api/codec.html#av.codec.context.ThreadType
+            for details.
+
+    """
+    container = av.open(url)
+    stream = container.streams.video[video_channel]
+    num_frames = stream.frames
+    # If number of frames not stored in the video file we have to decode all
+    # frames and count them.
+    if num_frames == 0:
+        stream.thread_type = thread_type
+        for _ in container.decode(stream):
+            num_frames += 1
+    return num_frames
+
+def all_video_frame_counts(
+    urls: List[str],
+    max_workers: int = None,
+    video_channel: int = 0,
+    thread_type: av.codec.context.ThreadType = av.codec.context.ThreadType.AUTO,
+) -> Tuple[int, List[int]]:
+    """Finds the number of frames in the videos at the given urls.
+
+    Videos are only decoded if no information about the number of frames is
+    stored in the video metadata.
+
+    Args:
+        urls:
+            A list of video urls.
+        max_workers:
+            Maximum number of workers. If `None` the number of workers is chosen 
+            based on the number of available cores.
+        video_channel:
+            The video stream channel from which to find the number of frames.
+        thread_type:
+            Which multithreading method to use for decoding the video.
+            See https://pyav.org/docs/stable/api/codec.html#av.codec.context.ThreadType
+            for details.
+
+    Returns:
+        A (total_frames, frame_counts) tuple where total_frames is the sum of 
+        all frames in all videos and frame_counts is a list with the number of
+        frames per video.
+
+    Raises:
+        RuntimeError:
+            If not all frame counts can be found for all videos.
+
+    """
+
+    def job(url):
+        try:
+            return utils.retry(
+                video_frame_count, 
+                url=url,
+                video_channel=video_channel,
+                thread_type=thread_type,
+            )
+        except RuntimeError:
+            return
+
+    with ThreadPoolExecutor(max_workers=max_workers) as executor:
+        frame_counts = list(executor.map(job, urls))
+        num_failed = 0
+        for count, url in zip(frame_counts, urls):
+            if count is None:
+                num_failed += 1
+                warnings.warn(
+                    f"Could not find the number of frames for video with "
+                    f"url: {url}"
+                )
+
+        if num_failed > 0:
+            raise RuntimeError(
+                f"Could not find the number of frames for {num_failed} videos!"
+            )
+
+        total_frames = sum(frame_counts)
+        return total_frames, frame_counts
```

Comment:
Why not only return the `frame_counts`? I think this would make the return type easier to understand. Summing it up can be easily done by the caller of the function if the total number of frames is needed.

Reply:
Good point! I changed it to only return frame_counts :)